From 4100417a7ee25bc3a261a38cc1752b464e8fd455 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Wed, 26 Feb 2020 18:25:42 +0100
Subject: [PATCH 1110/1239] arm64: dts: marvell: introduce armada-7040-db-F
 topology

The F topology of Armada-7040-DB setup enables using the
module with Peridot switch, working in unmanaged mode via
eth1 interface. Currently the mvpp2 NIC driver requires fixing
to support the fixed-link mode properly.

Update the documentation accordingly.

Change-Id: I664daa474480a17027f2e38af6b46808093b6d45
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/24004
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
---
 arch/arm/dts/Makefile                     |   1 +
 arch/arm/dts/armada-7040-db-F.dts         | 235 ++++++++++++++++++++
 doc/mvebu/a7k-a8k/armada70x0-db-setup.txt | 258 ++++++++++++----------
 3 files changed, 373 insertions(+), 121 deletions(-)
 create mode 100644 arch/arm/dts/armada-7040-db-F.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 8f9ab14988..7d16befb3f 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -200,6 +200,7 @@ dtb-$(CONFIG_ARCH_MVEBU) +=			\
 	armada-7040-db-C.dtb                    \
 	armada-7040-db-D.dtb			\
 	armada-7040-db-E.dtb			\
+	armada-7040-db-F.dtb			\
 	armada-8040-db.dtb			\
 	armada-8040-db-B.dtb			\
 	armada-8040-db-C.dtb			\
diff --git a/arch/arm/dts/armada-7040-db-F.dts b/arch/arm/dts/armada-7040-db-F.dts
new file mode 100644
index 0000000000..bba2cdf8da
--- /dev/null
+++ b/arch/arm/dts/armada-7040-db-F.dts
@@ -0,0 +1,235 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2020 Marvell International Ltd.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 7040 Development board platform
+ * Boot device: SPI NOR, 0x32 (SW3)
+ */
+
+#include "armada-70x0-db.dtsi"
+
+/ {
+	model = "Marvell Armada 7040 Development Board DB-88FA7040 setup(F)";
+	compatible = "marvell,armada7040-db", "marvell,armada7040",
+		     "marvell,armada-ap806-quad", "marvell,armada-ap806";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	aliases {
+		i2c0 = &cp0_i2c0;
+		spi0 = &cp0_spi1;
+		fuse0 = &ap_hd_efuse0;
+		fuse1 = &ap_ld_efuse0;
+		fuse2 = &ap_ld_efuse1;
+		fuse3 = &cp0_ld_efuse0;
+		fuse4 = &cp0_ld_efuse1;
+	};
+
+	memory@00000000 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+};
+
+&ap_pinctl {
+	   /* MPP Bus:
+	    * SDIO  [0-5]
+	    * UART0 [11,19]
+	    */
+		  /* 0 1 2 3 4 5 6 7 8 9 */
+	pin-func = < 1 1 1 1 1 1 0 0 0 0
+		     0 3 0 0 0 0 0 0 0 3 >;
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&ap_hd_efuse0 {
+	status = "okay";
+};
+
+&ap_ld_efuse0 {
+	status = "okay";
+};
+
+&ap_ld_efuse1 {
+	status = "okay";
+};
+
+&cp0_pcie2 {
+	status = "okay";
+};
+
+&cp0_i2c0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_i2c0_pins>;
+	status = "okay";
+	clock-frequency = <100000>;
+};
+
+&cp0_pinctl {
+		/* MPP Bus:
+		 * SPI   [13-16]
+		 * SATA1 [28]
+		 * UART0 [29-30]
+		 * SMI	 [32,34]
+		 * XSMI  [35-36]
+		 * I2C	 [37-38]
+		 * RGMII [44-55]
+		 * SD	 [56-61]
+		 * GPIO  [62]
+		 */
+		/*   0   1   2   3   4   5   6   7   8   9 */
+	pin-func = < 0   0   0   0   0   0   0   0   0   0
+		     0   0   0   3   3   3   3   0   0   0
+		     0   0   0   0   0   0   0   0   9   0xA
+		     0xA 0   7   0   7   7   7   2   2   0
+		     0   0   0   0   1   1   1   1   1   1
+		     1   1   1   1   1   1   0xE 0xE 0xE 0xE
+		     0xE 0xE 0>;
+};
+
+&cp0_spi1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_spi0_pins>;
+	reg = <0x700680 0x50>,		/* control */
+	      <0x2000000 0x1000000>,	/* CS0 */
+	      <0 0xffffffff>,		/* CS1 */
+	      <0 0xffffffff>,		/* CS2 */
+	      <0 0xffffffff>;		/* CS3 */
+	status = "okay";
+
+	spi-flash@0 {
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "jedec,spi-nor", "spi-flash";
+		reg = <0x0>;
+		spi-max-frequency = <108000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "U-Boot";
+				reg = <0x0 0x200000>;
+			};
+
+			partition@400000 {
+				label = "Filesystem";
+				reg = <0x200000 0xe00000>;
+			};
+		};
+	};
+};
+
+&cp0_ld_efuse0 {
+	status = "okay";
+};
+
+&cp0_ld_efuse1 {
+	status = "okay";
+};
+
+&cp0_sata0 {
+	status = "okay";
+};
+
+&cp0_usb3_0 {
+	status = "okay";
+};
+
+&cp0_usb3_1 {
+	status = "okay";
+};
+
+&cp0_comphy {
+	phy0 {
+		phy-type = <COMPHY_TYPE_SGMII1>;
+		phy-speed = <COMPHY_SPEED_3_125G>;
+	};
+
+	phy1 {
+		phy-type = <COMPHY_TYPE_USB3_HOST0>;
+	};
+
+	phy2 {
+		phy-type = <COMPHY_TYPE_SFI0>;
+		phy-speed = <COMPHY_SPEED_10_3125G>;
+	};
+
+	phy3 {
+		phy-type = <COMPHY_TYPE_SATA1>;
+	};
+
+	phy4 {
+		phy-type = <COMPHY_TYPE_USB3_HOST1>;
+	};
+
+	phy5 {
+		phy-type = <COMPHY_TYPE_PEX2>;
+	};
+};
+
+&cp0_utmi0 {
+	status = "okay";
+};
+
+&cp0_utmi1 {
+	status = "okay";
+};
+
+&ap_sdhci0 {
+	status = "okay";
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+};
+
+&cp0_sdhci0 {
+	status = "okay";
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+};
+
+&cp0_mdio {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+	};
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+&cp0_eth0 {
+	status = "okay";
+	phy-mode = "sfi"; /* lane-2 */
+};
+
+&cp0_eth1 {
+	status = "okay";
+	phy-mode = "sgmii-2500"; /* lane-0 */
+
+	fixed-link {
+		speed = <2500>;
+		full-duplex;
+	};
+};
+
+&cp0_eth2 {
+	status = "okay";
+	phy = <&phy1>;
+	phy-mode = "rgmii-id";
+};
diff --git a/doc/mvebu/a7k-a8k/armada70x0-db-setup.txt b/doc/mvebu/a7k-a8k/armada70x0-db-setup.txt
index fc306fa5f3..af11435cc5 100644
--- a/doc/mvebu/a7k-a8k/armada70x0-db-setup.txt
+++ b/doc/mvebu/a7k-a8k/armada70x0-db-setup.txt
@@ -27,90 +27,93 @@ Booting Linux requires using the same Device Tree configuration as used in U-Boo
 Board Setup Options
 ~~~~~~~~~~~~~~~~~~~
 
-+---------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|		| 1: A			| 2: B			| 3: C			| 4: D			| 4: E			|
-+===============+=======================+=======================+=======================+=======================+=======================+
-|Device tree	| armada-7040-db.dts	| armada-7040-db-B.dts	| armada-7040-db-C.dts	| armada-7040-db-D.dts  |armada-7040-db-E.dts	|
-+-----+---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-| CP0 | SW1:1	| ON			| OFF			| ON			| ON			| OFF                   |
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:2	| OFF			| ON			| OFF			| OFF			| OFF                   |
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:3	| ON			| ON			| ON			| ON			| ON                    |
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:4	| OFF			| ON			| OFF			| OFF			| OFF                   |
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:5	| OFF			| OFF			| OFF			| OFF			| OFF                   |
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:6	| ON			| ON			| ON			| ON			| ON                    |
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:7	| ON			| ON			| ON			| ON			| ON                    |
-|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|     | SW1:8	| OFF			| OFF			| OFF			| OFF			| OFF                   |
-+-----+---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
++---------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|		| 1: A			| 2: B			| 3: C			| 4: D			| 5: E			| 6: F			|
++===============+=======================+=======================+=======================+=======================+=======================+=======================+
+|Device tree	| armada-7040-db.dts	| armada-7040-db-B.dts	| armada-7040-db-C.dts	| armada-7040-db-D.dts  |armada-7040-db-E.dts	| armada-7040-db-F.dts	|
++-----+---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+| CP0 | SW1:1	| ON			| OFF			| ON			| ON			| OFF			| ON			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:2	| OFF			| ON			| OFF			| OFF			| OFF			| OFF			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:3	| ON			| ON			| ON			| ON			| ON			| ON			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:4	| OFF			| ON			| OFF			| OFF			| OFF			| OFF			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:5	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:6	| ON			| ON			| ON			| ON			| ON			| ON			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:7	| ON			| ON			| ON			| ON			| ON			| ON			|
+|     +---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|     | SW1:8	| OFF			| OFF			| OFF			| OFF			| OFF			| OFF			|
++-----+---------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
 
 DPR Setting
 ~~~~~~~~~~~
 
-+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-|			| 1: A			| 2: B			| 3: C			| 4: D			| 5: E			|
-+=======================+=======================+=======================+=======================+=======================+=======================+
-| Mpp Jumper Module	| DPR all = 1-2		| DPR all = 1-2		| DPR 133-143:		| DPR 133,135,136,137:	| DPR133-143:		|
-| (SLM1548)		|			|			|   2-3 position	|   N/C			|   1-2 position	|
-|			|			|			|			| DPR 134:		| DPR144-147:		|
-|			|			|			|			|   2-3 postion		|   2-3 position	|
-|			|			|			|			| DPR 138-143:		|			|
-|			|			|			|			|   2-3 postion		|			|
-+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-| Serdess Jumper Module	| DPR152,153,154,155:	| DPR152,153,154,155:	| DPR152,153,154,155:	| DPR152,153,154,155:	| DPR152,153,154,155:	|
-| (SLM1547)		|   1-2 position	|   1-2 position	|   1-2 position	|   1-2 position	|   1-2 position	|
-|			| DPR148,149,150,151:	| DPR148,149,150,151:	| DPR148,149,150,151:	| DPR148,149,150,151:	| DPR148,149,150,151:	|
-|			|   2-3 position	|   1-2 position	|   1-2 position	|   1-2 position	|   1-2 position	|
-+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-| CP0 Serdess module	| SW1:			| SW1:			|  SW1:			| sW1:			| SW1:			|
-| (SLM1521)		|   (1,3,6,7)= ON	|   (2,3,4,5,6,7)= ON	|    (1,3,6,7)= ON	|   (1,3,6,7)= ON	|   (3,6,7)= ON		|
-|			|   (2,4,5,8)= OFF	|   (1,8)=OFF		|    (2,4,5,8)=OFF	|   (2,4,5,8)= OFF	|   (1,2,4,5,8)=OFF	|
-|			| DPR2,3,4,5:		| DPR2,3,4,5:		| DPR2,3,4,5:		| DPR2,3,4,5:		| DPR2,3,4,5:		|
-|			|   1-2 position	|    2-3 position	|     1-2 position	|   1-2 position	|   1-2 position	|
-|			|			|			|			| Disconnect CON9 (USB3 |			|
-|			|			|			|			| host plug type A)	|			|
-|			|			|			|			| Connect CON11 (marvell|			|
-|			|			|			|			| pn CON641)		|			|
-|			|			|			|			| Connect R126,127,119,	|			|
-|			|			|			|			| 122, 121, 117		|			|
-+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-| Install TDM Module	|			|			| on DB-7040 CON46	|			|			|
-| (SLM1448)		|			|			|			|			|			|
-+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-| Install Audio Module	|			|			|			| on DB-7040 CON46	|			|
-| (SLM1449)		|			|			|			|			|			|
-+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
-| DB7040		|			|			|			| wire:			| DPR53-54: 1-2 position|
-|			|			|			|			|  from U36-1 to con47-2| DPR61: 2-3 position	|
-|			|			|			|			| DPR23 - NC		| DPR10-11 position 1-2	|
-|			|			|			|			|			| DPR12-15 position 1-2 |
-+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+|			| 1: A			| 2: B			| 3: C			| 4: D			| 5: E			| 6: F			|
++=======================+=======================+=======================+=======================+=======================+=======================+=======================+
+| Mpp Jumper Module	| DPR all = 1-2		| DPR all = 1-2		| DPR 133-143:		| DPR 133,135,136,137:	| DPR133-143:		|			|
+| (SLM1548)		|			|			|   2-3 position	|   N/C			|   1-2 position	|			|
+|			|			|			|			| DPR 134:		| DPR144-147:		|			|
+|			|			|			|			|   2-3 postion		|   2-3 position	|			|
+|			|			|			|			| DPR 138-143:		|			|			|
+|			|			|			|			|   2-3 postion		|			|			|
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+| Serdess Jumper Module	| DPR152,153,154,155:	| DPR152,153,154,155:	| DPR152,153,154,155:	| DPR152,153,154,155:	| DPR152,153,154,155:	| DPR152,153,154,155:	|
+| (SLM1547)		|   1-2 position	|   1-2 position	|   1-2 position	|   1-2 position	|   1-2 position	|   1-2 position	|
+|			| DPR148,149,150,151:	| DPR148,149,150,151:	| DPR148,149,150,151:	| DPR148,149,150,151:	| DPR148,149,150,151:	| DPR148,149,150,151:	|
+|			|   2-3 position	|   1-2 position	|   1-2 position	|   1-2 position	|   1-2 position	|   2-3 position	|
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+| CP0 Serdess module	| SW1:			| SW1:			|  SW1:			| sW1:			| SW1:			| SW1:			|
+| (SLM1521)		|   (1,3,6,7)= ON	|   (2,3,4,5,6,7)= ON	|    (1,3,6,7)= ON	|   (1,3,6,7)= ON	|   (3,6,7)= ON		|   (1,3,6,7)= ON	|
+|			|   (2,4,5,8)= OFF	|   (1,8)=OFF		|    (2,4,5,8)=OFF	|   (2,4,5,8)= OFF	|   (1,2,4,5,8)=OFF	|   (2,4,5,8)= OFF	|
+|			| DPR2,3,4,5:		| DPR2,3,4,5:		| DPR2,3,4,5:		| DPR2,3,4,5:		| DPR2,3,4,5:		| DPR2,3,4,5:		|
+|			|   1-2 position	|    2-3 position	|     1-2 position	|   1-2 position	|   1-2 position	|   1-2 position	|
+|			|			|			|			| Disconnect CON9 (USB3 |			|			|
+|			|			|			|			| host plug type A)	|			|			|
+|			|			|			|			| Connect CON11 (marvell|			|			|
+|			|			|			|			| pn CON641)		|			|			|
+|			|			|			|			| Connect R126,127,119,	|			|			|
+|			|			|			|			| 122, 121, 117		|			|			|
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+| Install TDM Module	|			|			| on DB-7040 CON46	|			|			|			|
+| (SLM1448)		|			|			|			|			|			|			|
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+| Install Audio Module	|			|			|			| on DB-7040 CON46	|			|			|
+| (SLM1449)		|			|			|			|			|			|			|
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+| DB7040		|			|			|			| wire:			| DPR53-54: 1-2 position|			|
+|			|			|			|			|  from U36-1 to con47-2| DPR61: 2-3 position	|			|
+|			|			|			|			| DPR23 - NC		| DPR10-11 position 1-2	|			|
+|			|			|			|			|			| DPR12-15 position 1-2 |			|
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
+| Install Switch Module	|			|			|			|			|			| on DB-7040 CON35/36	|
+| (SLM1449)		|			|			|			|			|			|			|
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+
 
 The tables below summarizes the interface configuration of each setup
 
 SerDes Configuration
 ~~~~~~~~~~~~~~~~~~~~
 
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| CP0 Lane	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		|
-+===============+===============+===============+===============+===============+===============+
-| 0		| SGMII1	| PICE0 (x4)	| PCIE0 (x1)    | PCIE0 (x1)    | PCIE0 (x1)	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| 1		| USB3_HOST0	| PCIE0 (x4)	| SATA0		| USB3_DEVICE0  | SATA0		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| 2		| SFI0 (10G)	| PCIE0 (x4)	| SFI0 (10G)    | SFI0 (10G)	| SGMII0 (1G)	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| 3		| SATA1		| PCIE0 (x4)	| SATA1		| SATA1         | SATA1		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| 4		| USB3_HOST1	| USB3_HOST1	| USB3_HOST1    | USB3_HOST1    | USB_HOST1	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| 5		| PCIE2 (x1)	| PCIE2 (x1)	| PCIE2 (x1)    | PCIE2 (x1)    | PCIE2 (x1)	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| CP0 Lane	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		| 6: F		|
++===============+===============+===============+===============+===============+===============+===============+
+| 0		| SGMII1	| PICE0 (x4)	| PCIE0 (x1)    | PCIE0 (x1)    | PCIE0 (x1)	| SGMII1	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| 1		| USB3_HOST0	| PCIE0 (x4)	| SATA0		| USB3_DEVICE0  | SATA0		| USB3_HOST0	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| 2		| SFI0 (10G)	| PCIE0 (x4)	| SFI0 (10G)    | SFI0 (10G)	| SGMII0 (1G)	| SFI0 (10G)	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| 3		| SATA1		| PCIE0 (x4)	| SATA1		| SATA1         | SATA1		| SATA1		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| 4		| USB3_HOST1	| USB3_HOST1	| USB3_HOST1    | USB3_HOST1    | USB_HOST1	| USB3_HOST1	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| 5		| PCIE2 (x1)	| PCIE2 (x1)	| PCIE2 (x1)    | PCIE2 (x1)    | PCIE2 (x1)	| PCIE2 (x1)	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
 
 - USB2_H0 (UTMI only)
 - USB2_H1 (UTMI & USB3 SerDes)
@@ -119,53 +122,53 @@ SerDes Configuration
 Multi-purpose pin configurations
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-+---------------+---------------+---------------+---------------+---------------+---------------+
-|AP806 pin	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		|
-+===============+===============+===============+===============+===============+===============+
-|	AP-SDIO | [0-5]		| [0-5]		| [0-5]         | [0-5]		| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-|	AP-SPI0 | N/C		| N/C		| N/C		| N/C		| [0-3]		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-|	AP-I2C	| N/C		| N/C		| N/C		| N/C		| [4,5]		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-|	AP-UART0| [11,19]	| [11,19]	| [11,19]       | [11,19]	| [11,19]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-
-+---------------+---------------+---------------+---------------+---------------+---------------+
-|CP0 pin:	| 1: A		| 3: B		| 3: C		| 4: D		| 5: E		|
-+===============+===============+===============+===============+===============+===============+
-| CP-SPI0	| N/C		| N/C		| [6-11]	| N/C		| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| CP-SPI1	| [13-16]	| [13-16]	| [13-16]	| N/C		| [13-16]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| NAND		| N/C		| N/C		| N/C		| [15-27]	| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| RGMII1	| [0-11]	| [0-11]	| N/C		| N/C		| [0-11]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| RGMII2	| N/C		| [44-55]	| [44-55]	| [44-55]	| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| CP-UART0	| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| CP-UART1	| N/C		| N/C		| N/C		| N/C		| [47,49-51]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| SD		| [56-61]	| [56-61]	| [56-61]	| [56-61]	| [54,56-61]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| GPIO		| [62]		| [62]		| [62]		| [62]		| [62]		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| TDM		| N/C		| N/C		| [0-5]		| N/C		| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| TWSI		| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| SATA0		| N/C		| N/C		| N/C		| N/C		| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| SATA1		| [28]		| [28]		| [28]		| [28]		| [28]		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| SMI		| [32,34]	| [32,34]	| [32,34]	| [32,34]	| [32,34]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| XSMI		| [35-36]	| [35-36]	| [35-36]	| [35-36]	| [35-36]	|
-+---------------+---------------+---------------+---------------+---------------+---------------+
-| USB_DEV	| N/C		| N/C		| N/C		| [7]		| N/C		|
-+---------------+---------------+---------------+---------------+---------------+---------------+
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|AP806 pin	| 1: A		| 2: B		| 3: C		| 4: D		| 5: E		| 6: F		|
++===============+===============+===============+===============+===============+===============+===============+
+|	AP-SDIO | [0-5]		| [0-5]		| [0-5]         | [0-5]		| N/C		| [0-5]		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|	AP-SPI0 | N/C		| N/C		| N/C		| N/C		| [0-3]		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|	AP-I2C	| N/C		| N/C		| N/C		| N/C		| [4,5]		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|	AP-UART0| [11,19]	| [11,19]	| [11,19]       | [11,19]	| [11,19]	| [11,19]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+|CP0 pin:	| 1: A		| 3: B		| 3: C		| 4: D		| 5: E		| 6: F		|
++===============+===============+===============+===============+===============+===============+===============+
+| CP-SPI0	| N/C		| N/C		| [6-11]	| N/C		| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| CP-SPI1	| [13-16]	| [13-16]	| [13-16]	| N/C		| [13-16]	| [13-16]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| NAND		| N/C		| N/C		| N/C		| [15-27]	| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| RGMII1	| [0-11]	| [0-11]	| N/C		| N/C		| [0-11]	| [0-11]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| RGMII2	| N/C		| [44-55]	| [44-55]	| [44-55]	| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| CP-UART0	| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	| [29-30]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| CP-UART1	| N/C		| N/C		| N/C		| N/C		| [47,49-51]	| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| SD		| [56-61]	| [56-61]	| [56-61]	| [56-61]	| [54,56-61]	| [56-61]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| GPIO		| [62]		| [62]		| [62]		| [62]		| [62]		| [62]		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| TDM		| N/C		| N/C		| [0-5]		| N/C		| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| TWSI		| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	| [37-38]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| SATA0		| N/C		| N/C		| N/C		| N/C		| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| SATA1		| [28]		| [28]		| [28]		| [28]		| [28]		| [28]		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| SMI		| [32,34]	| [32,34]	| [32,34]	| [32,34]	| [32,34]	| [32,34]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| XSMI		| [35-36]	| [35-36]	| [35-36]	| [35-36]	| [35-36]	| [35-36]	|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
+| USB_DEV	| N/C		| N/C		| N/C		| [7]		| N/C		| N/C		|
++---------------+---------------+---------------+---------------+---------------+---------------+---------------+
 
 Network configuration
 ~~~~~~~~~~~~~~~~~~~~~
@@ -222,8 +225,21 @@ Setup 5 (E)
 | eth1		| CP0   | eth1          | 2             | RGMII1                        |
 +---------------+-------+---------------+---------------+-------------------------------+
 
+Setup 6 (F):
+
++---------------+-------+---------------+---------------+-------------------------------+
+| Interface	| CP#	| PPv2 Port	| GMAC Port	| Board Interface		|
++===============+=======+===============+===============+===============================+
+| eth0		| CP0	| eth0		| 0		| 10G (Lane 2 connected to SFP) |
++---------------+-------+---------------+---------------+-------------------------------+
+| eth1		| CP0	| eth1		| 2		| SGMII1 (2.5G through switch	|
++---------------+-------+---------------+---------------+-------------------------------+
+| eth2		| CP0	| eth2		| 3		| RGMII2			|
++---------------+-------+---------------+---------------+-------------------------------+
+
 Notes:
 	- Equivalent of this configuration can be viewed in arch/arm64/boot/dts/mvebu/armada-cpn110.dtsi
 	- eth0/1/2/3 may in fact be higher numbers, if prior eth_x interfaces already exist.
+	- Setup F network via switch not operational in U-Boot.
 
 
-- 
2.29.0

